open-sphere-base/mantle/src/main/java/io/opensphere/mantle/data/geom/style/impl/AbstractLOBFeatureVisualizationStyle.java

```diff
@@ -799,11 +860,163 @@ private LOBRenderProperties determineRenderProperties(MapVisualizationInfo mapVi
         props.setDirectionalArrowLength(arrowLength);
         props = renderPropertyPool.getPoolInstance(props);
         return props;
     }
+
+    /**
+     * Adds error arcs to the set, if necessary.
+     *
+     * @param setToAddTo the set to which to add the geometries
+     * @param bd the builder data
+     * @param renderPropertyPool the render property pool
+     * @param lobGeom the LOB geometry
+     * @param centerLocation the center (point) location
+     */
+    private void addErrorArcs(Set<? super Geometry> setToAddTo, FeatureIndividualGeometryBuilderData bd,
+            RenderPropertyPool renderPropertyPool, LineOfBearingGeometry lobGeom, GeographicPosition centerLocation)
+    {
+        if (lobGeom != null && lobGeom.getRenderProperties().getLineLength() > 0)
+        {
+            double bearingError = getBearingError(bd.getMDP());
+            Length lengthError = getLobLengthError(bd.getMDP());
+            if (bearingError > 0 || lengthError.getMagnitude() > 0)
+            {
+                setToAddTo.add(createErrorArc(bd, renderPropertyPool, lobGeom, centerLocation, bearingError, lengthError));
+            }
+            if (lengthError.getMagnitude() > 0)
+            {
+                setToAddTo
+                        .add(createErrorArc(bd, renderPropertyPool, lobGeom, centerLocation, bearingError, lengthError.negate()));
+            }
+        }
+    }
+
+    /**
+     * Creates an error arc line geometry.
+     *
+     * @param bd the builder data
+     * @param renderPropertyPool the render property pool
+     * @param lobGeom the LOB geometry
+     * @param centerLocation the center (point) location
+     * @param bearingErrorDeg the bearing error in degrees
+     * @param legthError the length error
+     * @return the geometry
+     */
+    private PolylineGeometry createErrorArc(FeatureIndividualGeometryBuilderData bd, RenderPropertyPool renderPropertyPool,
+            LineOfBearingGeometry lobGeom, GeographicPosition centerLocation, double bearingErrorDeg, Length legthError)
+    {
+        // Create the line builder
+        PolylineGeometry.Builder<GeographicPosition> lineBuilder = new PolylineGeometry.Builder<>();
+        lineBuilder.setDataModelId(bd.getGeomId());
+        double lineLengthM = lobGeom.getRenderProperties().getLineLength() + legthError.inMeters();
+        if (bearingErrorDeg > 0)
+        {
+            // Create the ellipse arc builder
+            EllipseGeometry.ProjectedBuilder arcBuilder = new EllipseGeometry.ProjectedBuilder();
+            arcBuilder.setCenter(centerLocation);
+            arcBuilder.setAngle(lobGeom.getLineOrientation());
+            arcBuilder.setSemiMajorAxis(lineLengthM);
+            arcBuilder.setSemiMinorAxis(lineLengthM);
+            arcBuilder.setProjection(getToolbox().getMapManager().getProjection(Viewer3D.class).getSnapshot());
+            int vertexCount = (int)Math.ceil(bearingErrorDeg * 35 / 180 + 2);

```

Comment:
Most of the magic numbers in here make sense, but these are not obvious. What are they?